### PR TITLE
IS-364: Bump Golang CI.

### PR DESCRIPTION
**Pull request by metanerd**
## Summary
Bumping the Golang version for the build CI image. Additionally changed the naming scheme. 

#### Ticket Link
https://mattermost.atlassian.net/browse/IS-364

### CODE REVIEW

**Comment by agnivade** [comment]
There is a change in 1.14: https://tip.golang.org/doc/go1.14#net/url due to which a test is failing. I will fix it tomorrow. Rest looks good. Thanks !

File: build/README.md [hunk 1]
```diff
@@ -17,3 +17,4 @@ In our Docker Hub Repository we have the following images:
 - `mattermost/mattermost-build-server:dec-7-2018` which is based on Go 1.11 you can use for MM versions <= `5.8.0`
 - `mattermost/mattermost-build-server:feb-28-2019` which is based on Go 1.12 you can use for MM versions >= `5.9.0` <= `5.15.0`
 - `mattermost/mattermost-build-server:sep-17-2019` which is based on Go 1.12.9 you can use for MM versions >= `5.16.0`
+- `mattermost/mattermost-build-server:20200322_golang-1.14.1` which is based on Go 1.14.1 you can use for MM versions >= `5.22.0`
```
Comment (hanzei) [inline build/README.md]: Should go `1.14.1` used for 5.22 i.e. should this PR be cherry-picked in the release branch?

**Comment by hanzei** [comment]
Thanks for taking care of this @metanerd. It's great that we are staying up to date with the go releases.

File: build/README.md (hunk 1 above)
Comment (metanerd) [reply build/README.md]: Good catch!

**Comment by metanerd** [comment]
> Is there a PR for the enterprise repo?

Created. Checked everywhere else, but not enterprise m|

File: build/README.md [hunk 2]
```diff
@@ -17,3 +17,4 @@ In our Docker Hub Repository we have the following images:
 - `mattermost/mattermost-build-server:dec-7-2018` which is based on Go 1.11 you can use for MM versions <= `5.8.0`
 - `mattermost/mattermost-build-server:feb-28-2019` which is based on Go 1.12 you can use for MM versions >= `5.9.0` <= `5.15.0`
 - `mattermost/mattermost-build-server:sep-17-2019` which is based on Go 1.12.9 you can use for MM versions >= `5.16.0`
+- `mattermost/mattermost-build-server:20200322_golang-1.14.1` which is based on Go 1.14.1 you can use for MM versions >= `5.23.0`
```
Comment (hanzei) [inline build/README.md]: Just to double check: If go 1.14.1 should be used for `5.23.0`, this PR has to be cherry-picked one `5.22.0` is cut, right?

Comment (agnivade) [reply build/README.md]: No, we don't intend for 1.14.1 to be used in 5.23. Only from 5.24 onwards.

Comment (metanerd) [inline build/README.md]:
```suggestion
- `mattermost/mattermost-build-server:20200322_golang-1.14.1` which is based on Go 1.14.1 you can use for MM versions >= `5.24.0`
```

**Comment by agnivade** [comment]
@saturninoabril - This just needs some light smoke testing on the overall app. Nothing too deep. Just a basic check to see if everything works as expected. Thanks.

**Comment by saturninoabril** [comment]
Ready to merge.  Assigning to you @metanerd.

**Pull request merged**